geo/src/algorithm/densify_geodesic.rs

```diff
@@ -0,0 +1,243 @@
+{
+    type Output = MultiPolygon<f64>;
+
+    fn densify_geodesic(&self, max_distance: f64) -> Self::Output {
+        MultiPolygon::new(
+            self.iter().map(|polygon| polygon.densify_geodesic(max_distance)).collect(),
+        )
+    }
+}
+
+impl DensifyGeodesic<f64> for Polygon<f64>
+where
+    Line<f64>: GeodesicLength<f64>,
+    LineString<f64>: GeodesicLength<f64>,
+{
+    type Output = Polygon<f64>;
+
+    fn densify_geodesic(&self, max_distance: f64) -> Self::Output {
+        let densified_exterior = self.exterior().densify_geodesic(max_distance);
+        let densified_interiors = self.interiors().iter().map(|ring| ring.densify_geodesic(max_distance)).collect();
+        Polygon::new(densified_exterior, densified_interiors)
+    }
+}
+
+impl DensifyGeodesic<f64> for MultiLineString<f64>
+where
+    Line<f64>: GeodesicLength<f64>,
+    LineString<f64>: GeodesicLength<f64>,
+{
+    type Output = MultiLineString<f64>;
+
+    fn densify_geodesic(&self, max_distance: f64) -> Self::Output {
+        MultiLineString::new(
+            self.iter().map(|linestring| linestring.densify_geodesic(max_distance)).collect(),
+        )
+    }
+}
+
+impl DensifyGeodesic<f64> for LineString<f64>
+where
+    Line<f64>: GeodesicLength<f64>,
+    LineString<f64>: GeodesicLength<f64>,
+{
+    type Output = LineString<f64>;
+
+    fn densify_geodesic(&self, max_distance: f64) -> Self::Output {
+        if self.coords_count() == 0 {
+            return LineString::new(vec![]);
+        }
+
+        let mut new_line = vec![];
+        self.lines().for_each(|line| densify_line(line, &mut new_line, max_distance));
+        // we're done, push the last coordinate on to finish
+        new_line.push(self.points().last().unwrap());
+        LineString::from(new_line)
+    }
+}
+
+impl DensifyGeodesic<f64> for Line<f64>
+where
+    Line<f64>: GeodesicLength<f64>,
+    LineString<f64>: GeodesicLength<f64>,
+{
+    type Output = LineString<f64>;
+
+    fn densify_geodesic(&self, max_distance: f64) -> Self::Output {
+        let mut new_line = vec![];
+        densify_line(*self, &mut new_line, max_distance);
+        // we're done, push the last coordinate on to finish
+        new_line.push(self.end_point());
+        LineString::from(new_line)
+    }
+}
+
+impl DensifyGeodesic<f64> for Triangle<f64>
+where
+    Line<f64>: GeodesicLength<f64>,
+    LineString<f64>: GeodesicLength<f64>,
+{
+    type Output = Polygon<f64>;
+
+    fn densify_geodesic(&self, max_distance: f64) -> Self::Output {
+        self.to_polygon().densify_geodesic(max_distance)
+    }
+}
+
+impl DensifyGeodesic<f64> for Rect<f64>
+where
+    Line<f64>: GeodesicLength<f64>,
+    LineString<f64>: GeodesicLength<f64>,
+{
+    type Output = Polygon<f64>;
+
+    fn densify_geodesic(&self, max_distance: f64) -> Self::Output {
+        self.to_polygon().densify_geodesic(max_distance)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::coord;
+
```

Comment:
Can you add a test where `max_meters` is larger than each of the segment lengths?   i.e. where the algorithm is a no-op.